### PR TITLE
Adds metadata chunk, create upload session before uploading chunks

### PR DESCRIPTION
**Pull request by EdmundMai**
- Sends metadata as 0th chunk to alpha, last to beta (with chunk index == 0)
- Start at index 1 for file content chunks.
- Create upload session before sending chunks

Note: holding off on tests / BDD until after we get this stuff built out more

### CODE REVIEW

File: oyster/src/utils/file-processor.js [hunk 1]
```diff
@@ -1,16 +1,30 @@
 import _ from "lodash";
```
Comment (EdmundMai) [inline oyster/src/utils/file-processor.js]: This is the file where all the file upload magic happens in case you'd like to review @rfornea

**Comment by EdmundMai** [comment]
Note: About the comment above that Rebel made - Rebel, Aaron, and I discussed in the dev channel that the createUploadSession should provide the genesis_hash prior to uploading the chunks, so sending the metadata chunk last to the beta broker seems like expected behavior. (it will be easy to change if we're wrong)

File: oyster/src/utils/file-processor.js [hunk 2]
```diff
@@ -25,100 +39,134 @@ const createByteChunks = file => {
...
+        );
+
+        const reader = createReader(fileSlice => {
+          sendChunkToBroker(chunkIdx, fileSlice, handle).then(resolve);
+        });
+        reader.readAsBinaryString(blob);
+      })
+  );
+
+  return Promise.all(chunkRequests);
```
Comment (AaronVasquez) [inline oyster/src/utils/file-processor.js]: I think Promise.all has fail fast behavior, which we don't want. I also think Bruno mentioned he wants this throttled to like 10 concurrent requests at a time. Not going to block this PR, but we should add that in the future.

Comment (nymd) [reply oyster/src/utils/file-processor.js]: The request limiting to 10 is covered in another task, so should be fine

https://oyster.axosoft.com/viewitem?id=17&type=features&force_use_number=false

Comment (rfornea) [reply oyster/src/utils/file-processor.js]: Wonder if we could use observables/subjects or something if promises won't work?

Comment (AaronVasquez) [reply oyster/src/utils/file-processor.js]: +1 for Rx. Might be overkill for the testnet deadline, but adding it in the future would be good for async management.

For testnet we can do something simple like https://github.com/DJWassink/Promise-parallel-throttle. It solves our 2 problems.

**Pull request merged**